include/SimCore/Geo/ParserFactory.h

```diff
@@ -0,0 +1,61 @@
+#ifndef SIMCORE_GEO_PARSERFACTORY_H_
+#define SIMCORE_GEO_PARSERFACTORY_H_
+
+#include "SimCore/Geo/Parser.h"
+
+/*~~~~~~~~~~~~~~~~*/
+/*   C++ StdLib   */
+/*~~~~~~~~~~~~~~~~*/
+#include <map>
+
+/*~~~~~~~~~~~~~~~*/
+/*   Framework   */
+/*~~~~~~~~~~~~~~~*/
+#include "Framework/Configure/Parameters.h"
+
+namespace simcore {
+namespace geo {
+
+class ParserFactory {
+
+public:
+  /// Get the instance to this factory
+  static ParserFactory *getInstance();
```

Comment:
If keeping parser-specific factory, use reference instead of pointer. Less likely to get seg-vio behavior. Look at implementation of [PluginFactory](https://github.com/LDMX-Software/SimCore/blob/trunk/include/SimCore/PluginFactory.h) for how I used references.

Reply:
Thanks for the reminder here. I have been writing too much java code lately ;)